src/wallet/wallet.h

```diff
@@ -914,7 +914,13 @@ class CWallet final : public CCryptoKeyStore, public CValidationInterface
     void BlockDisconnected(const std::shared_ptr<const CBlock>& pblock) override;
     bool AddToWalletIfInvolvingMe(const CTransactionRef& tx, const CBlockIndex* pIndex, int posInBlock, bool fUpdate);
     int64_t RescanFromTime(int64_t startTime, const WalletRescanReserver& reserver, bool update);
-    CBlockIndex* ScanForWalletTransactions(CBlockIndex* pindexStart, CBlockIndex* pindexStop, const WalletRescanReserver& reserver, bool fUpdate = false);
+
+    enum class ScanResult : int {
+        SUCCESS = 0,
```

Comment:
nit: not sure if SUCCESS should use a non C-ish, non null positive value (1 probably)

Reply:
note that there is no need for an `enum class` to derive from int or even specify values for the members.